### DISCUSSION TITLE
converting openai tts to deepgram tts

### DISCUSSION THREAD

**Discussion by hboen1990** [deepgram/community#892] [General help] [answered]
Hi,
I want to try deepgram stt and tts.
I have the following working code when using OpenAI tts solution:
client = OpenAI(api_key=openai_api_key)

                        response =client.audio.speech.create(
                                                model='tts-1',
                                                voice='alloy',
                                                input=text,
                                                )
                        print('type(response.content)',type(response.content))
                        audio_base64 = base64.b64encode(response.content).decode('utf-8')
                        audio_html = f"""
                        <audio controls autoplay>
                            <source src="data:audio/wav;base64,{audio_base64}" type="audio/mp3">
                        </audio>
                        """

I did not found in the examples how to do it...
Any help would be gratefull.
Thanks in advance.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jpvajda** [comment, marked as answer]
@hboen1990  I recommend you take a look at our Starter Apps to see some working code examples of using Deepgram, they are a good getting started reference. 

https://github.com/deepgram-starters

**Reply by hboen1990** [reply]
Of course I did look at the examples. I found the documentation and examples are confusing, when u are new and starting with deepgram. Most of the examples are for sdk version 2 and I read that it will be deprecated and I installed version 3 but then most of the examples dont work. I read that version 4 is also avaiable....

If I run the following example "Chunked Text Source Payload" at https://developers.deepgram.com/docs/streaming-the-audio-output. The output.mp3 file is not recognized when I try to run...
The output.mp3 of the example: "Single Text Source Payload" is OK.

**Reply by hboen1990** [reply]
I am able to transform  the openai code to deepgram. I notice that _text to speech_ is "bad" when the text is NON english. The sound is more English than it should sound in the source language. I tested for Spanish, Dutch and Italian. If I compare the translated sound with openai or elevenlabs then they are much better.

**Comment by jpvajda** [comment]
@hboen1990  Which SDK are you trying to use specifically?

Depending on which example project you are looking at,  some might be outdated and no longer maintained. In Open Source we do our best to maintain as many projects as we can but unfortunately some just get out of date as we can't always keep up with all of them.
